test/libsolidity/util/ContractABIUtils.cpp

```diff
@@ -0,0 +1,87 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <test/libsolidity/util/ContractABIUtils.h>
+
+#include <liblangutil/Common.h>
+
+#include <boost/algorithm/string.hpp>
+
+#include <fstream>
+#include <memory>
+#include <regex>
+#include <stdexcept>
+
+using namespace dev;
+using namespace langutil;
+using namespace solidity;
+using namespace dev::solidity::test;
+using namespace std;
+using namespace soltest;
+
+dev::solidity::test::ParameterList ContractABIUtils::parametersFromJson(
+	Json::Value const& _contractABI,
+	string const& _functionName
+) const
+{
+	ParameterList abiParams;
+	for (auto const& function: _contractABI)
+		if (function["name"] == _functionName)
```

Comment:
Is it on purpose that this continues to iterate until it found all functions of that name? Or does the name include the signature?

Reply:
Ah, you're right. That's not how it should work. The signature is not being taken into account and the function also appends the types from all functions. But on the isoltest side, the correct types are used. I need to check that again.

Reply:
Fixed in #7052.